watcher.go

```diff
@@ -22,6 +22,26 @@ type Watcher struct {
 	once       sync.Once
 }
 
+type WatcherMetrics struct {
+	Name        string
+	LatencyMs   float64
+	LocalID     string
+	Channel     string
+	Protocol    string
+	Error       error
+	MessageSize int64
+}
+
+const (
+	RedisConnDoAuthMetric    = "RedisConnDoAuth"
+	RedisConnCloseMetric     = "RedisConnClose"
+	RedisDialMetric          = "RedisDial"
+	PubConnPublishMetric     = "PubConnPublish"
```

Comment:
How about the prefix for all these pub sub metrics be PubSub and remove the Conn part?  So PubSubPublish, PubSubReceive, PubSubSubscribe, PubSubUnsubscribe.

Reply:
@rwbarber2 fixed